contracts/utils/math/Math.sol

```diff
@@ -139,7 +139,9 @@ library Math {
     function mulDiv(uint256 x, uint256 y, uint256 denominator, Rounding rounding) internal pure returns (uint256) {
         uint256 result = mulDiv(x, y, denominator);
         if (rounding == Rounding.Up && mulmod(x, y, denominator) > 0) {
-            result += 1;
+            unchecked{
+                result += 1;
+            }
```

Comment:
when dealing with a counter that starts from 0 and increases slowly, it is true that an overflow is impossible in practice. It is however not the case here.

Result is not the result of a (slow) incrementation, but the result of a math operation. While i don't have an immediate counter example, I'm really not sure we can guarantee this won't overflow.